src/test/scala/GuessWhoGameSpec.scala

```diff
@@ -0,0 +1,162 @@
+import characters.{Person, Resources}
+import gameengine.GameEngine
+import org.scalatest.wordspec.AnyWordSpec
+
+import scala.collection.mutable.ListBuffer
+
+class GuessWhoGameSpec extends AnyWordSpec {
+
+  // Initialize game engine and resources once
+  val gameEngine: GameEngine = new GameEngine
+  val guessWhoGame: Resources = new Resources
+
+
+  // Initialize expected values directly
+  val expectedCharacters: List[Person] = guessWhoGame.charactersList
+  var expectedQuestions: List[String] = guessWhoGame.listOfQuestions
+
+  "Resources.returnListOfCharacters" should {
+    "return a list of characters" when {
+      "called to retrieve the characters list" in {
+        val characters = guessWhoGame.charactersList
+        assert(characters == expectedCharacters)
+      }
+    }
+  }
+
+  "Resources.returnListOfQuestions" should {
+    "return a list of questions" when {
+      "called to retrieve the questions list" in {
+        val questions = guessWhoGame.listOfQuestions
+        assert(questions == expectedQuestions)
+      }
+    }
+  }
+
+  "GameEngine.selectRandomCharacter" should {
+    "select a random character from the list" when {
+      "called to select a random character" in {
+        val characters = guessWhoGame.charactersList
+        val selectedCharacter = gameEngine.selectRandomCharacter(characters)
+        assert(characters.contains(selectedCharacter))
+      }
+    }
+  }
+
+  "GameEngine.selectRandomQuestion" should {
+    "select a random question from the list" when {
+      "called to select a random question" in {
+        val questions = guessWhoGame.listOfQuestions
+        val selectedQuestion = gameEngine.selectRandomQuestions()
+        assert(questions.contains(selectedQuestion))
+      }
+    }
+  }
+  //  "GameEngine.selectRandomQuestions" should {
+  //    "return 'no more questions' when the questions list is empty" in {
+  //      // Call one more time to get the 'no more questions' result
+  //      questionList.empty
+  //      val result = gameEngine.selectRandomQuestions()
+  //      assert(result == "no more questions")
+  //    }
+  //  }
+  "GameEngine.filterCharactersByAnswer" should {
+    "filter characters based on the string answer" when {
+      "called to filter characters by string answer" in {
+        val question = "Is your person male?"
+        val answer = false
+        val gameEngine = new GameEngine
+
```

Comment:
Good idea to make new GameEngine in each test. 
You could also do this  by writing: 
`"called to filter characters by string answer" in new GameEngine {...}`

Reply:
Thanks for the idea